### DISCUSSION TITLE
Add gzip support to API query

### DISCUSSION THREAD

**Discussion by mitch-karch** [grafana/grafana-infinity-datasource#779] [Q&A] [answered]
Adding `Accept-Encoding` of `gzip` would be incredibly beneficial as it allows for less data to be transmitted over networks unnecessarily. As this encoding is well understood and almost universally utilized, it would be fantastic to add support for gzip encoding as a header parameter for infinity.  Currently if you add the header manually, the resultant values cannot be parsed

**Comment by Tlunch** [comment]
This would be very helpful with egress charges as well.

**Comment by yesoreyeram** [comment, marked as answer]
https://github.com/grafana/grafana-infinity-datasource/pull/1068 should fix this

**Reply by mahendrapaipuri** [reply]
@yesoreyeram Cheers for this feature. Very helpful. So, if I understood well, #1068 supports gzip encoding for the requests made by Grafana backend server to the API resource. Is there a way to encode the responses of infinity plugin so that browser can load the data faster? Just to make it clear, my understanding of plugin working is as follows:

```browser <----> infinity <----> API resource```

So, gzip encoding is supported now on `infinity <----> API resource` but not supported (AFAIK) on `browser <----> infinity`. Is that correct? If so, is it something doable?

Cheers
